### DISCUSSION TITLE
How to fix this error "FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\gauth\\AppData\\Local\\Temp\\_MEI233082\\pyshadow\\resources\\querySelector.js"

### DISCUSSION THREAD

**Discussion by phonebotco** [pyinstaller/pyinstaller#6905] [PyInstaller] [answered]
Hi,

I am using Python 3.7.9 with Pyinstaller 5.0.1

I am trying to freeze my code which uses pyshadow.
When I execute it from Pycharm it works fine.
When I execute the exe after I freezed the code, I get this error:

`FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\gauth\\AppData\\Local\\Temp\\_MEI233082\\pyshadow\\resources\\querySelector.js`

So I add this file 'querySelector.js' in the freezing command:

`pyinstaller --onefile -F --uac-admin --icon="icon_PhoneBot_256.ico" --clean --noconsole  --add-data "querySelector.js;." main.py`

But I still get the same error.

Does anyone have any idea how to fix this issue?

**Comment by rokm** [comment, marked as answer]
The easiest way is to use `--collect-data pyshadow`. With `--add-data`, you need to collect the file into location where the code is expecting it (as opposed to top-level application directory), i.e., `--add-data path\to\pyshadow\resources\querySelector.js;pyshadow\resources`.

**Reply by phonebotco** [reply]
Thank for you so much.
